compaction/compaction_manager.cc

```diff
@@ -812,8 +812,6 @@ future<> compaction_manager::rewrite_sstables(replica::table* t, sstables::compa
             };
 
             auto maintenance_permit = co_await seastar::get_units(_maintenance_ops_sem, 1);
-            // Take write lock for table to serialize cleanup/upgrade sstables/scrub with major compaction/reshape/reshard.
-            auto write_lock_holder = co_await _compaction_state[&t].lock.hold_write_lock();
```

Comment:
what about taking read lock instead? it's not needed now, but to make it more future proof, e.g. guarantee serialization with a task type that takes write lock but doesn't go through maintenance_ops semaphore.

Reply:
I strongly oppose it.  Although acquiring the read lock would logically work, cpu cycles aren't free.
It's dead weight that makes the code slightly slower and makes it harder to understand, and it can be easily re-introduced if and when needed.

The code would be much easier to understand where each statement and expression has a clear purpose, vs. having to rely on uncertain future-proof plans to figure out why something like a lock is taken with no real need.

Reply:
You know what, I'll change it to a read lock for backporting purposes, but will get rid of it and all other `_compaction_state[t]` locks below maintenance_ops_sem other than major since it's really used only to synchronize major and regular compaction per-table.

Reply:
Fair enough. That's indeed the reason the rwlock was introduced for.